pkg/scheduler/schedule_one.go

```diff
@@ -88,13 +86,37 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	start := time.Now()
 	state := framework.NewCycleState()
 	state.SetRecordPluginMetrics(rand.Intn(100) < pluginMetricsSamplePercent)
+
 	// Initialize an empty podsToActivate struct, which will be filled up by plugins or stay empty.
 	podsToActivate := framework.NewPodsToActivate()
 	state.Write(framework.PodsToActivateKey, podsToActivate)
 
 	schedulingCycleCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	scheduleResult, err := sched.SchedulePod(schedulingCycleCtx, fwk, state, pod)
+	scheduleResult, assumedPodInfo, err := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, podsToActivate, start)
+	if err != nil {
+		// We don't need to do anything with a returned error here because it is properly handled in schedulingCycle().
+		return
+	}
+
+	// bind the pod to its host asynchronously (we can do this b/c of the assumption step above).
+	go func() {
+		bindingCycleCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		metrics.SchedulerGoroutines.WithLabelValues(metrics.Binding).Inc()
```

Comment:
It makes me think that we are not very diligent incrementing these counters throughout the codebase for everytime we create routines. Could you follow up on that? Maybe we should have a helper method to start routines.

Reply:
Sure.
I have questions about our goroutine strategy and the SchedulerGoroutines metric by the way.

- We only count SchedulerGoroutines here and the following place. https://github.com/kubernetes/kubernetes/blob/8e14585d8700d9cf7861be66943f023d11d41075/pkg/scheduler/schedule_one.go#L669-L673
But, actually, we often create goroutines with [parallelizer](https://github.com/kubernetes/kubernetes/blob/8e14585d8700d9cf7861be66943f023d11d41075/pkg/scheduler/framework/parallelize/parallelism.go). Should we inc/dec the SchedulerGoroutines metric in parallelizer as well?

- We don't restrict the number of goroutines created here and the following place. https://github.com/kubernetes/kubernetes/blob/8e14585d8700d9cf7861be66943f023d11d41075/pkg/scheduler/schedule_one.go#L669-L673
That means here and the following place can create goroutine more than [the parallelism option](https://github.com/kubernetes/kubernetes/blob/8e14585d8700d9cf7861be66943f023d11d41075/pkg/scheduler/scheduler.go#L157-L162). Should we do restrict the number of goroutines with the specified parallelism option value in these places as well?

Reply:
yes, we should